Week2/README.md

```diff
@@ -175,85 +188,95 @@ To learn more about this topic, check out the following:
 
 ```sql
-SELECT s.first_name, s.last_name, s.gender, s.grade, t.full_name
-FROM students s
-INNER JOIN teachers t
-ON s.teacher_number = t.teacher_number
+SELECT students.first_name, students.last_name, students.gender, students.grade, teachers.full_name
+FROM students
+INNER JOIN teachers
+ON students.teacher_number = teachers.teacher_number
 ```
 
-#### Self Joins
+The keyword `INNER` is optional and writing `JOIN teachers` will result in an inner join.
 
-A self join is a join in which a table is joined with itself (which is also called Unary relationships), especially when the table has a FOREIGN KEY which references its own PRIMARY KEY. To join a table itself means that each row of the table is combined with itself and with every other row of the table. In department, we want to get the employee's information with their direct manager. Here, each row in employees has a foreign key to itself as manager_id:
+#### Aliases
+
+When you join two tables there can be columns in both tables with the same name.
+To be explicit about the column you need you have to prefix the column with the table name like `table.column`
+If you want to type less you can create an alias for table names using `table AS alias`
 
 ```sql
-SELECT a.full_name AS full_name, b.full_name AS manager_name
-FROM employee a, employee b
-WHERE a.manager_id = b.id;
+SELECT teachers.*, tq.title
+FROM teachers
+JOIN teacher_qualifications AS `tq` ON tq.teacher_id = teacher.id
 ```
 
+The `AS` keyword is optional, so you could write it also like `teacher_qualifications tq`
+
 #### Right and Left Joins
 
 If we wanted to simply append information about teachers to our students table regardless of whether a student has a teacher or not, we would use a left join. A left join returns all records from table A and any matching records from table B.
 
 ```sql
-SEKECT s.first_name, s.last_name, t.full_name
-FROM studetns s
-LEFT JOIN teachers t
-ON s.teacher_number = t.teacher_number
+SELECT students.first_name, students.last_name, teachers.full_name
+FROM students
+LEFT JOIN teachers ON students.teacher_number = teachers.number
 ```
 
-It can be reversed. The reverse way of querying is called RIGHT JOIN.
+If you tried to keep all the teachers and wanted to see teachers with or without students you need a right join.
+The right join will keep all records from the joined table and discard records, from the preceding tables, that can't be joined
 
 Check out the following to get a more visual idea of what `joins` are:
 
 - [Understand SQL Joins in 10 Minutes](https://www.youtube.com/watch?v=tvMGoxmQzgQ)
 - [Joins in SQL](https://www.youtube.com/watch?v=efpFCd8iFAQ)
 
+Other types of joins are less common but can sometimes be needed.
+If you're interested you can read more about [self joins](https://www.w3schools.com/sql/sql_join_self.asp) and [full outer joins](https://www.w3schools.com/sql/sql_join_full.asp).
```

Comment:
Instead of w3school link, we can provide the employee and manger join example here.
It is a better example of self join.

Reply:
In my opinion self joins and full outer joins aren't subjects students need to know about. But I added it back in to comply with the request from Rob (in the thread about self joins) 
> add things that we feel they should know about but won't be using too often to an 'extra reading' part.

Keeping it in the extra curriculum part seemed like a good compromise.